src/lib/support/ThreadOperationalDataset.h

```diff
@@ -247,6 +254,36 @@ class OperationalDataset
      */
     void UnsetPSKc(void);
 
+    /**
+     * Returns ByteSpan pointing to the channel mask within the dataset.
+     * @retval CHIP_NO_ERROR on success.
+     * @retval CHIP_ERROR_TLV_TAG_NOT_FOUND if the channel mask is not present in the dataset.
+     * @retval CHIP_ERROR_INVALID_TLV_ELEMENT if the TLV element is invalid.
+     */
+    CHIP_ERROR GetChannelMask(ByteSpan & aChannelMask) const;
```

Comment:
Should talk here a bit about how long that ByteSpan will be valid.

Should this just take a MutableByteSpan and copy the value out?

Reply:
I can add some documentation about the life time. I think in general we shouldn't need to be copying out in these accessors (e.g. `GetExpandedPanIdAsByteSpan` doesn't copy out, though all the original ones do). Those array references are also super clunky to work with, I'd kinda like the fixed sizes ones to just return a FixedByteSpan... but I didn't want to completely rewrite this class.

Reply:
Done